### PR TITLE
Restructure and convert to Maven

### PR DESCRIPTION
**Pull request by CyberShadow**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: java
+
+dist: trusty
```
Comment (ScoreUnder) [inline .travis.yml]: Trusty is pretty old by now, any reason not to build on xenial?

Comment (JoshDreamland) [reply .travis.yml]: Hey, long time, no see. There's no compelling reason to use Trusty, but there's no advantage to our projects to upgrading to Xenial, and Bionic isn't available on Travis, as far as I'm aware.

**Comment by RobertBColton** [comment]
Closing as I have recreated this in #395 to fix the merge conflicts here.

**Comment by RobertBColton** [comment]
@CyberShadow I need some additional assistance before I can merge this, I am getting stuck in a number of locations for several reasons.

First, in the `.travis.yml` you are manually building the JoshEdit jar it seems. I think we would want that automated by the POM. We don't want to have to manually copy joshedit.jar to build LGM locally. We would accept a POM file for JoshEdit in its upstream repo.

Second, although the Travis build passes, it doesn't seem to be actually picking up any classes.
https://travis-ci.org/IsmAvatar/LateralGM/builds/520516431#L1252

My pull request was submitted from branch `CyberShadow-maven` on this repo so I could edit it. If you git pull, you should be able to commit to it with me so we can both work on the pr.

**Pull request closed without merging**